api/utils/token/chained_validator.go

```diff
@@ -0,0 +1,28 @@
+package token
+
+import (
+	"context"
+	"errors"
+)
+
+type ChainedValidator struct{ validators []ValidatorInterface }
+
+var _ ValidatorInterface = &ChainedValidator{}
+var errNoIssuersFound = errors.New("no issuers found")
+
+func NewChainedValidator(validators ...ValidatorInterface) *ChainedValidator {
+	return &ChainedValidator{validators}
+}
+
+func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (principal TokenPrincipal, err error) {
+	for index, validator := range v.validators {
+		principal, err = validator.ValidateToken(ctx, token)
+		if err == nil {
+			return principal, nil
+		} else if index == len(v.validators)-1 {
+			return nil, err
+		}
+	}
```

Comment:
Optional suggestion - to make it easier to read
```suggestion
		if principal, err = validator.ValidateToken(ctx, token); err == nil {
			return principal, nil
		} else if index == len(v.validators)-1 {
			return nil, err
		}
```
or
```suggestion
		principal, err = validator.ValidateToken(ctx, token)
		if err == nil {
			return principal, nil
		} 
		if index == len(v.validators)-1 {
			return nil, err
		}
```
or 
```suggestion
	var principal TokenPrincipal
	var err error
	for _, validator := range v.validators {
		if principal, err = validator.ValidateToken(ctx, token); err == nil {
			return principal, nil
		}
	}
	if err != nil {
	  return nil, err
	}
```

Reply:
fixed with:

```golang
func (v *ChainedValidator) ValidateToken(ctx context.Context, token string) (TokenPrincipal, error) {
	var errs []error

	for _, validator := range v.validators {
		principal, err := validator.ValidateToken(ctx, token)
		if principal != nil {
			return principal, nil
		}
		errs = append(errs, err)
	}

	return nil, fmt.Errorf("%w: %v", errNoIssuersFound, errors.Join(errs...))
}
```